MathLib/LinAlg/MatrixProviderUser.h

```diff
@@ -0,0 +1,79 @@
+/**
+ * \copyright
```

Comment:
A brief induction about the provider is needed. For example, it provides instances of global matrix for the system of linear equations.